doc/source/debug.rst

```diff
@@ -0,0 +1,191 @@
+tells you that they are experiencing strange behavior. Let's take a look
+at our deployment to figure out what is going on.
+
+.. note::
+
+   For our real world scenario, we'll assume that our Kubernetes namespace
+   is called ``jhub``. Your namespace may be called something different
+
+``kubectl get pod``
+~~~~~~~~~~~~~~~~~~~
+To list all pods in your Kubernetes deployment::
+
+    kubectl get pod --namespace==jhub
+
+This will output a list of all pods being used in the deployment.
+
+**Real world scenario:** In our case, we see two pods for the JupyterHub
+infrastructure (``hub-deployment`` and ``proxy-deployment``) as well as one user
+pod that was created when somebody logged in to the JupyterHub.
+
+Here's an example of the output::
+
+    $ kubectl --namespace=jhub get pod
+    NAME                                READY     STATUS         RESTARTS   AGE
+    hub-deployment-3311438805-xnfvp     1/1       Running        0          2m
+    jupyter-choldgraf                   0/1       ErrImagePull   0          25s
+    proxy-deployment-1227971824-mn2wd   1/1       Running        0          5h
+
+Here we can see the two JupyterHub pods, as well as a single user pod. Note
+that all user pods will begin with ``jupyter-``.
+
+In particular, keep an eye on the ``STATUS`` column. If a given
+pod contains something other than ``Running``, then something may be wrong.
+
+In this case, we can see that our user's pod is in the ``ErrImagePull`` state.
+This generally means that there's something wrong with the Docker image that
+is defined in ``singleuser`` in our helm chart. Let's dig further...
+
+``kubectl describe pod``
+~~~~~~~~~~~~~~~~~~~~~~~~
+To see more detail about the state of a specific pod, use the following
+command::
+
+    kubectl --namespace=jhub describe pod <POD_NAME>
+
+This will output several pieces of information, including configuration and
+settings for the pod. The final section you'll see is a list of recent
+events. These can be particularly informative, as often an error will
+show up in this section.
+
+**Real world scenario:** In our case, one of the lines in the events page
+displays an error::
+
+      $ kubectl --namespace=jhub describe pod jupyter-choldgraf
+      ...
+      2m            52s             4       kubelet, gke-jhubtest-default-pool-52c36683-jv6r        spec.containers{notebook}       Warning         Failed           Failed to pull image "jupyter/scipy-notebook:v0.4": rpc error: code = 2 desc = Error response from daemon: {"message":"manifest for jupyter/scipy-notebook:v0.4 not found"}
+      ...
+
+It seems there is indeed something wrong with the Docker image. Let's confirm
+this by getting another view on the events that have transpired in the pod.
+
+``kubectl logs``
+~~~~~~~~~~~~~~~~
+If you only want to see the latest logs for a pod, use the following command::
+
+    kubectl --namespace=jhub logs <POD_NAME>
+
+This will output a list of the recent events for the pod. Parse these logs

```

Comment:
This will show you the logs from the pod, which often contain useful information about what is going wrong.